cmd/root.go

```diff
@@ -529,30 +511,169 @@ status code.`)
 	localFlags.BoolVar(&c.conf.PSC, "psc", false,
 		"(*) Connect to the PSC endpoint for all instances")
 
-	v := viper.NewWithOptions(viper.EnvKeyReplacer(strings.NewReplacer("-", "_")))
-	v.SetEnvPrefix(envPrefix)
-	v.AutomaticEnv()
-	// Ignoring the error here since its only occurence is if one of the pflags
-	// is nil which is never the case here.
-	_ = v.BindPFlags(globalFlags)
-	_ = v.BindPFlags(localFlags)
-
-	// Override any unset flags with Viper values to use the pflags
-	// object as a single source of truth.
-	localFlags.VisitAll(func(f *pflag.Flag) {
+	return c
+}
+
+func loadConfig(c *Command) error {
+	v, err := getViper(c)
+	if err != nil {
+		return err
+	}
+
+	_ = v.BindPFlags(c.Flags())
+
+	c.Flags().VisitAll(func(f *pflag.Flag) {
+		// Override any unset flags with Viper values to use the pflags
+		// object as a single source of truth.
 		if !f.Changed && v.IsSet(f.Name) {
 			val := v.Get(f.Name)
-			_ = localFlags.Set(f.Name, fmt.Sprintf("%v", val))
+			_ = c.Flags().Set(f.Name, fmt.Sprintf("%v", val))
 		}
 	})
-	globalFlags.VisitAll(func(f *pflag.Flag) {
-		if !f.Changed && v.IsSet(f.Name) {
-			val := v.Get(f.Name)
-			_ = globalFlags.Set(f.Name, fmt.Sprintf("%v", val))
+
+	return nil
+}
+
+func loadArgs(c *Command, args []string) error {
+	v, err := getViper(c)
+	if err != nil {
+		return err
+	}
+
+	// If args is not already populated, try to read from the environment.
+	if len(args) == 0 {
+		args = instanceFromEnv(args)
+	}
+
+	// If no environment args are present, try to read from the config file.
+	if len(args) == 0 {
+		args = instanceFromConfigFile(v)
+	}
+
+	// Handle logger separately from config
+	if c.conf.StructuredLogs {
+		c.logger, c.cleanup = log.NewStructuredLogger(c.conf.Quiet)
+	}
+
+	if c.conf.Quiet {
+		c.logger = log.NewStdLogger(io.Discard, os.Stderr)
+	}
+
+	err = parseConfig(c, c.conf, args)
+	if err != nil {
+		return err
+	}
+
+	// The arguments are parsed. Usage is no longer needed.
+	c.SilenceUsage = true
+
+	// Errors will be handled by logging from here on.
+	c.SilenceErrors = true
+
+	return nil
+}
+
+func getViper(c *Command) (*viper.Viper, error) {
```

Comment:
How about `initViper`? Cf. https://go.dev/doc/effective_go#Getters

Reply:
changed name to initViper.